src/org/opendatakit/briefcase/ui/reused/source/ShowSourceForm.java

```diff
@@ -32,26 +34,54 @@
   private JLabel sourceLabel;
   private JButton resetButton;
   private JLabel actionLabel;
+  private JButton reloadButton;
   private List<Runnable> onResetCallbacks = new ArrayList<>();
+  private List<Runnable> onReloadCallbacks = new ArrayList<>();
+  private boolean showReloadButton;
 
-  private ShowSourceForm(String action) {
+  private ShowSourceForm(String action, boolean showReloadButton) {
     $$$setupUI$$$();
     this.action = action;
+    this.showReloadButton = showReloadButton;
 
     resetButton.addActionListener(__ -> onResetCallbacks.forEach(Runnable::run));
+    reloadButton.addActionListener(__ -> onReloadCallbacks.forEach(Runnable::run));
+
+    onReloadCallbacks.add(() -> {
+      reloadButton.setEnabled(false);
+      TimerTask task = new TimerTask() {
+        public void run() {
+          reloadButton.setEnabled(true);
+        }
+      };
+      Timer timer = new Timer();
+      long delayMillis = 5000L;
+      timer.schedule(task, delayMillis);
+    });
+    if (!showReloadButton)
+      reloadButton.setVisible(false);
+  }
```

Comment:
If we *were* to use a timer here, using a javax.swing.Timer instead would have advantages.
<img width="1639" alt="screen shot 2018-08-26 at 3 38 28 pm" src="https://user-images.githubusercontent.com/55099/44633906-3e21af80-a946-11e8-953d-963999a175f6.png">

https://docs.oracle.com/javase/tutorial/uiswing/misc/timer.html